pkg/api/graph/graph.go

```diff
@@ -404,6 +447,10 @@ func (g Graph) SubgraphWithNodes(nodes []graph.Node, fn EdgeFunc) Graph {
 	for _, node := range nodes {
 		out.internal.AddNode(node)
 	}
+	fmt.Println("Edges under review:")
+	for _, edge := range g.internal.Edges() {
```

Comment:
I spent a couple of hours trying to debug this. `g.internal.Edges()` returns a fixed number of edges non-deterministically (http://pastebin.test.redhat.com/299528). @deads2k can you have a look at this?